tests/unit/src/kakarot/precompiles/test_blake2f.cairo

```diff
@@ -0,0 +1,555 @@
+    assert input[120] = 0x00;
+    assert input[121] = 0x00;
+    assert input[122] = 0x00;
+    assert input[123] = 0x00;
+    assert input[124] = 0x00;
+    assert input[125] = 0x00;
+    assert input[126] = 0x00;
+    assert input[127] = 0x00;
+    assert input[128] = 0x00;
+    assert input[129] = 0x00;
+    assert input[130] = 0x00;
+    assert input[131] = 0x00;
+    assert input[132] = 0x00;
+    assert input[133] = 0x00;
+    assert input[134] = 0x00;
+    assert input[135] = 0x00;
+    assert input[136] = 0x00;
+    assert input[137] = 0x00;
+    assert input[138] = 0x00;
+    assert input[139] = 0x00;
+    assert input[140] = 0x00;
+    assert input[141] = 0x00;
+    assert input[142] = 0x00;
+    assert input[143] = 0x00;
+    assert input[144] = 0x00;
+    assert input[145] = 0x00;
+    assert input[146] = 0x00;
+    assert input[147] = 0x00;
+    assert input[148] = 0x00;
+    assert input[149] = 0x00;
+    assert input[150] = 0x00;
+    assert input[151] = 0x00;
+    assert input[152] = 0x00;
+    assert input[153] = 0x00;
+    assert input[154] = 0x00;
+    assert input[155] = 0x00;
+    assert input[156] = 0x00;
+    assert input[157] = 0x00;
+    assert input[158] = 0x00;
+    assert input[159] = 0x00;
+    assert input[160] = 0x00;
+    assert input[161] = 0x00;
+    assert input[162] = 0x00;
+    assert input[163] = 0x00;
+    assert input[164] = 0x00;
+    assert input[165] = 0x00;
+    assert input[166] = 0x00;
+    assert input[167] = 0x00;
+    assert input[168] = 0x00;
+    assert input[169] = 0x00;
+    assert input[170] = 0x00;
+    assert input[171] = 0x00;
+    assert input[172] = 0x00;
+    assert input[173] = 0x00;
+    assert input[174] = 0x00;
+    assert input[175] = 0x00;
+    assert input[176] = 0x00;
+    assert input[177] = 0x00;
+    assert input[178] = 0x00;
+    assert input[179] = 0x00;
+    assert input[180] = 0x00;
+    assert input[181] = 0x00;
+    assert input[182] = 0x00;
+    assert input[183] = 0x00;
+    assert input[184] = 0x00;
+    assert input[185] = 0x00;
+    assert input[186] = 0x00;
+    assert input[187] = 0x00;
+    assert input[188] = 0x00;
+    assert input[189] = 0x00;
+    assert input[190] = 0x00;
+    assert input[191] = 0x00;
+    assert input[192] = 0x00;
+    assert input[193] = 0x00;
+    assert input[194] = 0x00;
+    assert input[195] = 0x00;
+    assert input[196] = 0x03;
+    assert input[197] = 0x00;
+    assert input[198] = 0x00;
+    assert input[199] = 0x00;
+    assert input[200] = 0x00;
+    assert input[201] = 0x00;
+    assert input[202] = 0x00;
+    assert input[203] = 0x00;
+    assert input[204] = 0x00;
+    assert input[205] = 0x00;
+    assert input[206] = 0x00;
+    assert input[207] = 0x00;
+    assert input[208] = 0x00;
+    assert input[209] = 0x00;
+    assert input[210] = 0x00;
+    assert input[211] = 0x00;
+    assert input[212] = 0x01;
+
+    // When
+    let (output_len, output, gas) = PrecompileBlake2f.run(213, input);
+
+    // Then
```

Comment:
Could you provide a short comment linking how we can be sure that `blake2f(input) == output` for a "battle-tested" implementation of blake2f?
Some kind of way to diff test?

Reply:
I tried blake2b from hashlib and pycryptodome but for some reason I wasn't getting the correct hash (must have something to do with the internal state). 

One possible way would be to use py-evm which provides in its utils the access to the [blake2b compression function](https://github.com/ethereum/py-evm/blob/master/eth/_utils/blake2/compression.py). If that works for you I will update the test file for the precompile.

Reply:
you can also do a check using the evm.codes playground

Reply:
since you've added `py-evm` and compare against another `blake2f`, do we still need this?
Also don't forget to add an test case in `test_cases` (the one from evm.playground for example). to make sure that the precompile as correctly called when calling its address